python/cudf/cudf/dataframe/numerical.py

```diff
@@ -50,10 +50,15 @@ def __contains__(self, item):
         """
         item_found = False
         try:
-            if self.find_first_value(item):
+            if (
+                cudautils.find_first(
+                    self.astype("float_").data.mem, float(item)
+                )
+                != -1
+            ):
                 item_found = True
-        except ValueError:
-            """This means value not found"""
+        except Exception:
+            "Nothing to be done"
```

Comment:
What type of Exception are we trying to catch here? If we're failing to encode `item` in a way Numba can understand then we should catch that above here. I fear this could hide Numba failing to execute and silently returning an incorrect answer.

Reply:
I think, I don't have to use any `try` over here, as I was using it along with find_first_value.